lib/saluki-components/src/sources/checks/python_exposed_modules.rs

```diff
@@ -0,0 +1,160 @@
+use tracing::trace;
+
+use super::*;
+
+/// submit_metric is called from the AgentCheck implementation when a check submits a metric.
+/// Python signature:
+///     aggregator.submit_metric(self, self.check_id, mtype, name, value, tags, hostname, flush_first_value)
+///
+/// TODO(remy): should mtype be a PyMetricType?
+#[allow(clippy::too_many_arguments)]
+#[pyfunction]
+#[pyo3(pass_module)]
+pub(crate) fn submit_metric(
+    module: &Bound<PyModule>, _class: PyObject, _check_id: String, mtype: i32, name: String, value: f64,
+    tags: Vec<String>, hostname: String, _flush_first_value: bool,
+) {
+    trace!(
+        "submit_metric called with name: {}, value: {}, tags: {:?}, hostname: {}",
+        name,
+        value,
+        tags,
+        hostname
+    );
+
+    let check_metric = CheckMetric {
+        name,
+        metric_type: mtype.into(),
+        value,
+        tags,
+    };
+
+    match module.getattr("SUBMISSION_QUEUE") {
+        Ok(py_item) => match py_item.extract::<Py<python_scheduler::PythonSenderHolder>>() {
+            Ok(q) => {
+                let res = pyo3::Python::with_gil(|py| q.bind_borrowed(py).borrow_mut().sender.clone());
```

Comment:
I think this is a bit convoluted, if you understood the pyo3 borrow system better you could end up with some cleaner code. IIRC @GeorgeHahn actually had a suggestion here at one point

Reply:
My previous suggestion is lost to the sands of time, but looking at this with fresh eyes makes me wonder if the runtime GIL acquisition can be replaced with a function parameter. The docs [here](https://docs.rs/pyo3/latest/pyo3/marker/struct.Python.html#obtaining-a-python-token) make it sound like that's possible and faster.